recipe/meta.yaml

```diff
@@ -29,40 +26,32 @@ build:
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
-    - {{ compiler('cxx') }}  # [not win]
-    - {{ stdlib('c') }}
-    - clang                  # [win]
-    - clangxx                # [win]
   host:
-    - python
+    - python {{ python_min }}
     - pip
     - setuptools
-    - numpy
   run:
-    # To ensure ABI compatibility, we install the same C++
-    # compiler that was used for building.
-    - {{ cxx_compiler }}_{{ target_platform }}  # [unix]
-    - clang                  # [win]
-    - clangxx                # [win]
-    - python
-    - {{ pin_compatible('numpy') }}
+    - gxx                    # [linux]
+    - clangxx                # [osx or win]
```

Comment:
In #77, @rgommers commented yesterday:
> there should not be a compiler dependency on any platform. Pythran is fundamentally a transpiler, and when you want to use it with a C++ compiler that can be MinGW just as well as Clang-cl on Windows, hence the hard `clang` dependency is counterproductive.

I don't have a strong opinion here, but given that this PR seems squarely aimed at that issue, we should discuss IMO (and please cross-reference related discussions/issues like this from the start going forward).

Reply:
Ah great, I missed this PR, and I guess @isuruf missed the old issue. 

This PR seems like a significant improvement already, at least allowing installation on free-threaded CPython (and avoiding lots of unnecessary CI time, etc.). So happy to see it merged even as is.

That said, the compiler dependencies here should go in the `test` section in my opinion. They are actively unhelpful as direct dependencies.

Reply:
> That said, the compiler dependencies here should go in the test section in my opinion. They are actively unhelpful as direct dependencies.

Yeah, let's keep that discussion to another PR.